doc/source/tutorials/add_model.rst

```diff
@@ -0,0 +1,327 @@
+    :animate: fade-in
+
+    * Get familiar with fairseq2 basics (:ref:`basics-design-philosophy`)
+    * Understanding of fairseq2 assets system (:ref:`basics-assets`)
+    * Ensure you have fairseq2 installed
+
+Overview
+--------
+
+fairseq2 uses a model system that makes it easy to add support for new models.
+There are two main scenarios:
+
+1. **Adding a new model to an existing family** (most common) - When you want to add a new size or variant of an existing model architecture
+2. **Creating an entirely new model family** (advanced) - When you need to implement a completely new model architecture
+
+This tutorial focuses on the first scenario, which covers 95% of use cases. For the second scenario, refer to the existing model family implementations as reference.
+
+Understanding Model Families
+----------------------------
+
+In fairseq2, a **model family** groups related model architectures that share the same underlying implementation but differ in size or configuration. For example:
+
+- **Qwen family**: ``qwen25_3b``, ``qwen25_3b``, ``qwen25_14b``, ``qwen3_8b``, etc.
+- **LLaMA family**: ``llama3_8b``, ``llama3_70b``, ``llama3_2_1b``, etc.
+- **Mistral family**: ``mistral_7b``, ``mistral_8x7b``, etc.
+
+Each family consists of:
+
+- **Model configurations** (architectures): Define structural parameters (layers, dimensions, etc.)
+- **Asset cards**: YAML files specifying download locations and metadata
+- **Model implementation**: The actual PyTorch model code and loading logic
+
+Step-by-Step Guide: Adding a Model to Existing Family
+-----------------------------------------------------
+
+Let's walk through adding ``qwen25_3b_instruct`` to the existing Qwen family.
+
+Step 1: Add Model Architecture Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Model architectures are defined as configuration presets that specify the structural parameters of your model.
+
+1. **Navigate to the model family's config file:**
+
+   .. code-block:: bash
+
+       src/fairseq2/models/qwen/config.py
+
+2. **Add a new architecture function:**
+
+   .. code-block:: python
+
+       @arch("qwen25_3b_instruct")
+       def qwen25_3b_instruct() -> QwenConfig:
+           """Configuration for Qwen2.5-3B-Instruct model.
+           """
+           config = QwenConfig()
+
+           # Set model dimensions and structure
+           config.model_dim = 2048
+           ...
+
+           return config
+
+   **Key points:**
+
+   - The ``@arch`` decorator registers the configuration with the given name
+   - The function name should match or describe the architecture
+   - Use the appropriate config class for the model family (``QwenConfig`` for Qwen models)
+   - Set all necessary parameters for your specific model variant
+
+Step 2: Create Asset Card
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Asset cards are YAML files that tell fairseq2 where to find your model checkpoints and how to load them.
+
+1. **Navigate to the model family's asset card file:**
```

Comment:
Similar to above, ideally we should show how users can do this without touching fairseq2:

```python
def setup_my_fairseq2_extension(container: DependencyContainer) -> None:
  register_package_assets("my_package.assets")
  # or
  # register_file_assets("path/to/assets")
```

And put the asset card to the corresponding location.